pkg/controllers/jobset_controller.go

```diff
@@ -377,15 +416,39 @@ func (r *JobSetReconciler) createJobs(ctx context.Context, js *jobset.JobSet, ow
 			return err
 		}
 	}
-
+	getReplicatedJobStatus := func(replicatedJobStatus []jobset.ReplicatedJobStatus, replicatedJobName string) jobset.ReplicatedJobStatus {
+		for _, val := range replicatedJobStatus {
+			if val.Name == replicatedJobName {
+				return val
+			}
+		}
+		return jobset.ReplicatedJobStatus{}
+	}
 	var lock sync.Mutex
 	var finalErrs []error
-
-	for _, rjob := range js.Spec.ReplicatedJobs {
+	for idx, rjob := range js.Spec.ReplicatedJobs {
 		jobs, err := constructJobsFromTemplate(js, &rjob, ownedJobs)
 		if err != nil {
 			return err
 		}
+		jobStatus := getReplicatedJobStatus(replicatedJobStatus, rjob.Name)
+		if !ptr.Deref(js.Spec.Suspend, false) {
```

Comment:
can we have a `jobSetSuspended` func instead of `ptr.Deref(js.Spec.Suspend, false)` which is easy to misread.